src/main/java/org/spongepowered/common/mixin/core/entity/merchant/villager/AbstractVillagerEntityMixin.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.core.entity.merchant.villager;
+
+import net.minecraft.entity.merchant.villager.AbstractVillagerEntity;
+import net.minecraft.entity.merchant.villager.VillagerEntity;
+import net.minecraft.entity.merchant.villager.VillagerTrades;
+import net.minecraft.item.MerchantOffer;
+import net.minecraft.item.MerchantOffers;
+import org.spongepowered.api.data.type.Profession;
+import org.spongepowered.api.entity.living.trader.Villager;
+import org.spongepowered.api.item.merchant.TradeOffer;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.common.mixin.core.entity.AgeableEntityMixin;
+import org.spongepowered.common.registry.SpongeVillagerRegistry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Mixin(AbstractVillagerEntity.class)
+public abstract class AbstractVillagerEntityMixin extends AgeableEntityMixin {
+    private List<MerchantOffer> impl$newOffers = new ArrayList<>();
+
+    /**
+     * @author i509VCB - February 17th, 2020 - 1.14.4
+     *
+     * @reason Here we apply the Trade Mutators. For a villager we will apply our mutators and then add them to the villager's merchant offers.
+     * If a wandering trader is the subject here, we do nothing since that merchant would already have it's new merchant offers.
+     *
+     * @todo: Do we want to support Trade Mutators on Wandering Trader's TradeOffers?
```

Comment:
Yes, but that can be put up as an issue to remind us later.

Reply:
Made an api issue about it